ast/src/parsed/mod.rs

```diff
@@ -140,9 +140,18 @@ impl PilStatement {
                         .map(move |v| (name, Some(&v.name), SymbolCategory::TypeConstructor)),
                 ),
             ),
-            PilStatement::TraitDeclaration(_, TraitDeclaration { name, .. }) => {
-                Box::new(once((name, None, SymbolCategory::TraitDeclaration)))
-            }
+            PilStatement::TraitDeclaration(
+                _,
+                TraitDeclaration {
+                    name, functions, ..
+                },
+            ) => Box::new(
+                once((name, None, SymbolCategory::TraitDeclaration)).chain(
+                    functions
+                        .iter()
+                        .map(move |f| (name, Some(&f.name), SymbolCategory::TypeConstructor)),
```

Comment:
I think `Value` should be more suitable. The main difference between TypeConstructor and Value is that TypeConstructor can be used inside patterns - but trait functions should not be usable inside patterns.

Reply:
Oh, I see, thanks!